server.js

```diff
@@ -18,7 +36,137 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Hello! This is the backend-part of a previous project called Happy Thoughts.");
+});
+
+app.get('/thoughts', async (req, res) => { 
+//V1 Mongoose   
+  const { page } = req.query;
+  try {
+    const thoughts = await Thought.find({})
+    .sort({createdAt: -1})
+    .skip((page -1) * 20).limit(20);
+    res.status(200).json({success: true, response: thoughts});
+  } catch (error) {
+    res.status(400).json({success: false, response: error});
+  }
+});
+
+app.get('/thoughts/:thoughtId', async (req, res) => {
+  try {
+    const fetchId = await Thought.findOne({_id: req.params.thoughtId});
+    if(fetchId) {
+      res.status(200).json({
+        data: fetchId,
+        success: true,
+      })
+    } else {
+      res.status(404).json({
+        error: 'ID not foud ',
+        success: false,
+      })
+    }
+  } catch (err) {
+    res.status(400).json({
+      error: 'Invalid ID',
+      success: false,
+    })
+  }
+})
+
+//V2 Mongo with query
+// const { page, perPage, numPage = +page, numPerPage = +perPage } = req.query;
+// try {
+//   const thoughts = await Thought.aggregate([
+//     {
+//       $sort: {
+//         createdAt: -1
+//       }
+//     },
+//     {
+//       $skip: ( numPage - 1) * numPerPage
+//     },
+//     {
+//       $limit: numPerPage
+//     }
+//   ]);
+
+//   res.status(200).json({success: true, response: thoughts});
+// } catch (error) {
+//   res.status(400).json({success: false, response: error});
+// }
+// });
+
+app.post('/thoughts', async (req, res) => {
+  const { message } = req.body;
+
+  try {
+    const newThought = await new Thought({ message: message}).save()
+    res.status(201).json({
+      response: newThought,
+      success: true
+    });
+  } catch (error) { 
+    res.status(400).json({
+      response: 'Could not save message',
+      success: false
+    });
+  }
+});
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params;
+
+  try {
+    const updatedLikes = await Thought.findByIdAndUpdate(thoughtId, {$inc: {hearts: 1}});
+    res.status(200).json({
+      response: updatedLikes,
```

Comment:
If you want you can add a little message to the user in order to confirm the like that has been updated,  it would look something like this: `({response: Thought like ${updatedLikes.message} has been increased, success: true)}`  - but with the tics before and after the message text in order to get it as a template literal - did not know how to add those here in order to get a code block :-)